### PR TITLE
Expand isGlobalTSFile + test

### PR DESCRIPTION
**Pull request by tomv564**
Two fixes to allow global typings to load content as documented in #220

As content for these files in not fetched as part of resolving references, the typescript compiler is unable to resolve anything from these typings and will log a diagnostic:
`File '.../node_modules/tslib/tslib.d.ts' is not a module.`

The regex for global typings installed in `node_modules/@typings` was restricted to the 'node' subfolder, this should include all content in @typings.

The TSlib dependency is created in the compiler when the `importHelpers` option is true:
https://github.com/Microsoft/TypeScript/blob/2150a7796b0dfe83c5286755410e0fd1646ed86c/src/compiler/program.ts#L1264-L1273

### CODE REVIEW

File: src/test/util-test.ts [hunk 1]
```diff
@@ -35,4 +35,12 @@ describe('util', () => {
 			assert.equal(matches, true);
 		});
 	});
+	describe('isGlobalTSFile()', () => {
+		it('should match declaration files in node_modules', () => {
+			assert.equal(isGlobalTSFile('/node_modules/@types/not_node/not_node.d.ts'), true);
```
Comment (felixfbecker) [inline src/test/util-test.ts]: This screws the semantics a bit, `isGlobalTSFile` is supposed to be a heuristic whitelist of known global declarations. Most declarations in `@types` are _not_ global. These should be fetched, but not by changing the definition of `isGlobalTSFile`

Comment (tomv564) [inline src/test/util-test.ts]: Ah, right these should match another `import` elsewhere. 
I will remove this case and better document cases where the pre-loading fails (if it still does).

The tsdlib.d.ts is not referenced in source (only ProjectConfiguration can know from checking options.importHelpers) - is it okay if it stays in the isGlobalTSFile check?

Comment (felixfbecker) [inline src/test/util-test.ts]: I guess yes since it is global and common

**Comment by felixfbecker** [comment]
Wanna add mocha in here?

**Comment by tomv564** [comment]
Not by itself, no. Isn't whitelisting individual packages a losing race?
I like the idea of fetching all files up-front, then we can perhaps do away with isGlobalTSFile altogether.

**Comment by felixfbecker** [comment]
It is a compromise. We should definitely fetch all files, but that can take a few seconds. We want to be able to respond to a hover or j2d in the first seconds after init without fetching all files. Given most typings are in external module format nowadays, `@types/node` + `@types/mocha` would probably cover 99% of used global typings.

**Comment by tomv564** [comment]
@felixfbecker added mocha, jest and jasmine to isGlobalTS + tests. Want to keep it to this change for now?

**Pull request merged**